### PR TITLE
download checkstyle-8.0-all.jar when it not exists

### PR DESCRIPTION
**Pull request by jdneo**
work item: #20 

download

### CODE REVIEW

File: client/src/extension.ts [hunk 1]
```diff
@@ -53,7 +58,7 @@ namespace Configuration {
             }
             result.push({
                 autocheck: config.get<boolean>('autocheck'),
```
Comment (Eskibear) [inline client/src/extension.ts]: better way to represent `path.join(__dirname, '..', '..'` ? seems a little ugly

File: server/src/server.ts [hunk 2]
```diff
@@ -75,7 +75,7 @@ enum ConfigurationType {
 
 const defaultSettings: ICheckStyleSettings = {
     autocheck: true,
-    jarPath: path.join(__dirname, '..', 'resources', 'checkstyle-8.5-all.jar'),
```
Comment (Eskibear) [inline server/src/server.ts]: same here, here only ONE `'..'`, a little bit confusing, maybe set default jarPath as a constant string?

File: README.md [hunk 3]
```diff
@@ -23,7 +23,7 @@ This extension provides several commands in the Command Palette (F1 or Ctrl + Sh
 ```
 
 * ```checkstyle.autocheck``` - Specify whether the checkstyle extension will check Java files automatically.
-* ```checkstyle.jarPath``` - Path to the checkstyle jar file. By default, the extension will use [checkstyle-8.5-all.jar](https://sourceforge.net/projects/checkstyle/files/checkstyle/8.5/)
+* ```checkstyle.jarPath``` - Path to the checkstyle jar file. By default, the extension will use [checkstyle-8.0-all.jar](https://sourceforge.net/projects/checkstyle/files/checkstyle/8.0/)
 * ```checkstyle.configurationFile``` - Specify the checkstyle configuration file. Entering ```google_checks``` will use [google_checks.xml](https://raw.githubusercontent.com/checkstyle/checkstyle/master/src/main/resources/google_checks.xml). Entering ```sun_checks``` will use [sun_checks.xml](https://raw.githubusercontent.com/checkstyle/checkstyle/master/src/main/resources/google_checks.xml). Or you can specify the configuration file path in your local machine. By default, the extension will use ```google_checks```.
```
Comment (adashen) [inline README.md]: why version is downgraded?

Comment (Eskibear) [reply README.md]: @adashen he just follows eclipse to use a most commonly used version by default.

File: client/src/extension.ts [hunk 4]
```diff
@@ -115,3 +122,14 @@ export function deactivate(): Thenable<void> {
     Configuration.dispose();
     return client.stop();
 }
+
+async function ensureDefaultJarInstalled(outputChannel: OutputChannel, version: string = '8.0'): Promise<void> {
+    const extPath: string = extensions.getExtension('shengchen.vscode-checkstyle').extensionPath;
```
Comment (adashen) [inline client/src/extension.ts]: extensionPath should be able to got from context?

File: client/src/extension.ts [hunk 5]
```diff
@@ -115,3 +122,14 @@ export function deactivate(): Thenable<void> {
...
+    if (await pathExists(path.join(extPath, 'resources', `checkstyle-${version}-all.jar`))) {
```
Comment (adashen) [inline client/src/extension.ts]: the if block could combine with else as if(! await pathExists)

File: README.md (hunk 3 above)
Comment (jdneo) [reply README.md]: Exactly as @Eskibear said.

File: client/src/extension.ts (hunk 4 above)
Comment (jdneo) [reply client/src/extension.ts]: Emmmm, I want to get the extension installation path here. Does context have the API for this?

Comment (jdneo) [reply client/src/extension.ts]: After investigation, I find it can be get from context.

File: server/src/server.ts (hunk 2 above)
Comment (jdneo) [reply server/src/server.ts]: Filed an issue here: #18

File: client/src/extension.ts (hunk 1 above)
Comment (jdneo) [reply client/src/extension.ts]: #18

**Pull request merged**